app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java

```diff
@@ -1,44 +1,104 @@
+import com.mapbox.mapboxsdk.utils.BitmapUtils;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class MapFragment extends Fragment {
 
     private MapView mapView;
+    private MapboxMap mapboxMap;
+    private MarkerSymbolManager markerSymbolManager;
+    private final String SYMBOL_ID = "MARKER_MAP";
+    private final String BUNDLE_ID = "bundleT";
+    private final String MARKERS_ID = "markers_size";
 
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater,
                              @Nullable ViewGroup container,
                              @Nullable Bundle savedInstanceState) {
-        Mapbox.getInstance(getContext().getApplicationContext(), getString(R.string.mapbox_access_token));
-
+        Mapbox.getInstance(getContext(), getString(R.string.mapbox_access_token));
         View view = inflater.inflate(R.layout.fragment_map, container, false);
 
-        mapView = (MapView) view.findViewById(R.id.mapView);
+        mapView = view.findViewById(R.id.mapView);
         mapView.onCreate(savedInstanceState);
-        mapView.getMapAsync(mapboxMap -> mapboxMap.setStyle(Style.MAPBOX_STREETS, style -> {
+        mapView.getMapAsync(mapboxMap -> {
+            this.mapboxMap = mapboxMap;
+
+            mapboxMap.setStyle(Style.MAPBOX_STREETS, style -> {
+                initializeMarkerSymbolManager(style);
+            });
 
-        }));
+            mapboxMap.addOnMapClickListener(point -> {
+                markerSymbolManager.removeAllMarker();
+                markerSymbolManager.addMarker(point);
+                return false;
+            });
+        });
         return view;
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        System.out.println(savedInstanceState);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        mapView.onResume();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        mapView.onDestroy();
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        mapView.onPause();
+    }
+
+    @Override
+    public void onLowMemory() {
+        super.onLowMemory();
+        mapView.onLowMemory();
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        mapView.onSaveInstanceState(outState);
+    }
 
+    private void initializeMarkerSymbolManager(@NonNull Style styleOnLoaded) {
+        Drawable marker = ContextCompat.getDrawable(getContext(), R.drawable.ic_marker_map);
+        styleOnLoaded.addImage(SYMBOL_ID, BitmapUtils.getBitmapFromDrawable(marker));
+        SymbolManager symbolManager = new SymbolManager(mapView, mapboxMap, styleOnLoaded);
+        markerSymbolManager = new MarkerSymbolManager(symbolManager, SYMBOL_ID);
```

Comment:
I would write
```suggestion
        this.markerSymbolManager = new MarkerSymbolManager(symbolManager, SYMBOL_ID);
```
just to be more explicit (but that's just really my way of doing it).

Reply:
I agree